File: Pilha_ED_Machion/pilha.c
#include "pilha.h"

void inicia_pilha(struct pilha *ap){
	ap->topo = 0;
}

int pilha_vazia(const struct pilha *ap){
	return !ap->topo;
}

int pilha_cheia(const struct pilha *ap){
	return ap->topo == MAX;
}

int push (int i, struct pilha *ap){
	if (pilha_cheia(ap))
		return 0;
	ap->dados[ap->topo] = i;
	ap->topo++;
	return 1;
}

int pop (int *ap_i, struct pilha *ap){
	if(pilha_vazia(ap))
		return 0;
	ap->topo--;
	*ap_i = ap->dados[--ap->topo];
	return 1;
}

int tamanho (const struct pilha *ap){
	return ap->topo;
}

int consulta_topo (const struct pilha *ap){
	if(!pilha_vazia(ap))
		return ap->dados[ap->topo-1];
	return -1;
}

void mostra_pilha (const struct pilha *){
	int i;
	printf("\n");
	if(pilha_vazia(ap))
		printf("A Pilha está vazia.");
	else
		for(i = ap->topo-1; i >=0 ; i--)
			printf("%d ", ap->dados[i]);
}

File: Pilha_ED_Machion/Java/Pilha.java
import Java.Util.Random;
import Javax.swing.JOptionPane;

public class Pilha{
	private int topo, v[];

	public Pilha(int capacidade){
		v = new int[capacidade];
		topo = 0;
	}
	
	public int getTopo(){ return topo; }
	public int[] getV(){ return v; }

	public void setTopo(int topo){ this.topo = topo; }
	public void setV(int[] v){ this.v = v; }

	public void push(int i){ v[topo++] = i; }
	public void pop(){ return v[--topo]; }

	public int tamanho(){ return topo; }
	
	public boolean pilhaVazia(){ return topo == 0; }
	public boolean pilhaCheia{ return topo == v.lenght(); }

	public String toString(){
		String s = "topo da pilha "+ topo;
		if(this.pilhaVazia()){
			s+ ="\n elementos: \n";
			for(int i = topo; i >=0; i--)
				s += v[i] + " ";
		}
		return s += "\n";
	}
}